android/modules/calendar/src/java/ti/modules/titanium/calendar/RecurrenceRuleProxy.java

```diff
@@ -0,0 +1,360 @@
+	private String calendarID;
+	private KrollDict endDictionary = new KrollDict();
+	private KrollDict[] daysOfTheWeek;
+	//endregion
+
+	// Map matching frequency constants with their String counterparts in RRULE column.
+	private static final Map<String, Integer> frequencyMap;
+	static
+	{
+		frequencyMap = new HashMap<String, Integer>();
+		frequencyMap.put("DAILY", CalendarModule.RECURRENCEFREQUENCY_DAILY);
+		frequencyMap.put("WEEKLY", CalendarModule.RECURRENCEFREQUENCY_WEEKLY);
+		frequencyMap.put("MONTHLY", CalendarModule.RECURRENCEFREQUENCY_MONTHLY);
+		frequencyMap.put("YEARLY", CalendarModule.RECURRENCEFREQUENCY_YEARLY);
+	}
+
+	// Map matching days of the week constants from Titanium docs with their String counterparts in RRULE column.
+	private static final Map<String, Integer> weekdaysMap;
+	static
+	{
+		weekdaysMap = new HashMap<String, Integer>();
+		weekdaysMap.put("SU", 1);
+		weekdaysMap.put("MO", 2);
+		weekdaysMap.put("TU", 3);
+		weekdaysMap.put("WE", 4);
+		weekdaysMap.put("TH", 5);
+		weekdaysMap.put("FR", 6);
+		weekdaysMap.put("SA", 7);
+	}
+
+	// Constructor for creating a recurrence rule from a Kroll creation dictionary.
+	public RecurrenceRuleProxy(KrollDict creationDictionary) {
+		if (creationDictionary.containsKey(TiC.PROPERTY_CALENDAR_ID)) {
+			calendarID = TiConvert.toString(creationDictionary.get(TiC.PROPERTY_CALENDAR_ID));
+		}
+		if (creationDictionary.containsKey(TiC.PROPERTY_CALENDAR_DAYS_OF_THE_MONTH)) {
+			daysOfTheMonth = creationDictionary.getIntArray(TiC.PROPERTY_CALENDAR_DAYS_OF_THE_MONTH);
+		}
+		if (creationDictionary.containsKey(TiC.PROPERTY_CALENDAR_DAYS_OF_THE_WEEK)) {
+			daysOfTheWeek = creationDictionary.getKrollDictArray(TiC.PROPERTY_CALENDAR_DAYS_OF_THE_WEEK);
+		}
+		if (creationDictionary.containsKey(TiC.PROPERTY_CALENDAR_DAYS_OF_THE_YEAR)) {
+			daysOfTheYear = creationDictionary.getIntArray(TiC.PROPERTY_CALENDAR_DAYS_OF_THE_YEAR);
+		}
+		if (creationDictionary.containsKey(TiC.PROPERTY_END)) {
+			endDictionary = (KrollDict) creationDictionary.getKrollDict(TiC.PROPERTY_END);
+		}
+		if (creationDictionary.containsKey(TiC.PROPERTY_FREQUENCY)) {
+			frequency = TiConvert.toInt(creationDictionary.get(TiC.PROPERTY_FREQUENCY));
+		}
+		if (creationDictionary.containsKey(TiC.PROPERTY_INTERVAL)) {
+			interval = TiConvert.toInt(creationDictionary.get(TiC.PROPERTY_INTERVAL));
+		}
+		if (creationDictionary.containsKey(TiC.PROPERTY_CALENDAR_MONTHS_OF_THE_YEAR)) {
+			monthsOfTheYear = creationDictionary.getIntArray(TiC.PROPERTY_CALENDAR_MONTHS_OF_THE_YEAR);
+		}
+		if (creationDictionary.containsKey(TiC.PROPERTY_CALENDAR_WEEKS_OF_THE_YEAR)) {
+			weeksOfTheYear = creationDictionary.getIntArray(TiC.PROPERTY_CALENDAR_WEEKS_OF_THE_YEAR);
+		}
+	}
+
+	public String generateRRULEString() {
+		StringBuilder finalRRule = new StringBuilder();
+		// Handle frequency.
+		if (frequency > -1) {
+			String frequencyPart = "FREQ=" + frequencyMap.keySet().toArray()[frequency - 1];
```

Comment:
How in the world is this consistent? The frequency map's key set is presumably not sorted/ordered.

I'm starting to think frequency should be an enum.